core/src/main/java/com/digitalpebble/stormcrawler/protocol/file/FileProtocol.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Licensed to DigitalPebble Ltd under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * DigitalPebble licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.digitalpebble.stormcrawler.protocol.file;
+
+import org.apache.storm.Config;
+
+import com.digitalpebble.stormcrawler.Metadata;
+import com.digitalpebble.stormcrawler.protocol.Protocol;
+import com.digitalpebble.stormcrawler.protocol.ProtocolResponse;
+import com.digitalpebble.stormcrawler.protocol.RobotRulesParser;
+import com.digitalpebble.stormcrawler.util.ConfUtils;
+
+import crawlercommons.robots.BaseRobotRules;
+
+public class FileProtocol implements Protocol {
+
+    private int maxContentLength;
+    private String encoding;
+    private boolean crawlParent;
+
+    @Override
+    public void configure(Config conf) {
+        encoding = ConfUtils.getString(conf, "file.encoding", "UTF-8");
+        maxContentLength = ConfUtils.getInt(conf, "file.content.limit",
+                64 * 1024);
+        crawlParent = ConfUtils.getBoolean(conf, "crawlParent", true);

```

Comment:
not sure about this one. if users want to crawl the parent dir, they would give it as seed straight away. It should not be set to true by default + I'd rather have something called `file.crawl.parent` so that it is obivous that the parameter is related to the file protocol.